artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/mirror/AMQPMirrorControllerSource.java

```diff
@@ -467,6 +504,28 @@ public static void validateProtocolData(ReferenceIDSupplier referenceIDSupplier,
       }
    }
 
+   /**
+    * Checks if the message ref should be filtered or not.
+    * @param ref the message to filter
+    * @return true if the INTERNAL_RECEIVER_ID_FILTER annotation of the message is set to a different value
+    * that the remoteMirrorID, false otherwise.
```

Comment:
(No need to keep the suggestion commit, its just easier to show the suggested change this way)

```suggestion
    * @param ref the message to check
    * @return true if the INTERNAL_RECEIVER_ID_FILTER annotation of the message is set to a different value
    * than the remoteMirrorID, false otherwise.
```